concreate/descriptor.py

```diff
@@ -73,13 +74,25 @@ def merge(self, descriptor):
         Args:
           descriptor - a concreate descritor
         """
+        if self.descriptor is descriptor:
+            return
         try:
             self.descriptor = tools.merge_dictionaries(
                 self.descriptor, descriptor)
         except KeyError as ex:
             logger.debug(ex, exc_info=True)
             raise ConcreateError("Cannot merge descriptors, see log message for more information")
 
+    def _process_artifacts(self):
+        """ Processes descriptor artifacts section and generate default
+        value 'name' for each artifact which doesnt have 'name' specified.
+        """
+        artifacts = {}
+        for artifact in self.descriptor['artifacts']:
+            resource = Resource.new(artifact, self.directory)
+            artifacts[resource.name] = resource
+        self.descriptor['artifacts'] = artifacts
```

Comment:
descriptor should not contain any object different to dictionary - we should build proper objects for everything or nothing.

Reply:
Agreed, but that's quite a large change and descriptor is an object already.